runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java

```diff
@@ -142,14 +148,15 @@ public CorfuCompileProxy(CorfuRuntime rt, UUID streamID, Class<T> type, Object[]
     ) {
         this.rt = rt;
         this.streamID = streamID;
+        this.streamName = streamName;
         this.type = type;
         this.args = args;
         this.serializer = serializer;
 
         // Since the VLO is thread safe we don't need to use a thread safe stream implementation
         // because the VLO will control access to the stream
         underlyingObject = new VersionLockedObject<T>(this::getNewInstance,
-                new StreamViewSMRAdapter(rt, rt.getStreamsView().getUnsafe(streamID)),
+                new StreamViewSMRAdapter(rt, rt.getStreamsView().getUnsafe(streamID), streamName),
```

Comment:
Why don't you store the stream reference (i.e. `rt.getStreamsView().getUnsafe(streamID)`) and then just use `stream.getName()`

Reply:
See above